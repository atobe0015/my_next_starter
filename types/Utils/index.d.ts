import React from 'react'

export type ForwardRefTypeOfElement<
  T extends keyof JSX.IntrinsicElements,
  U = unknown
> = React.ForwardRefExoticComponent<
  React.PropsWithRef<JSX.IntrinsicElements[T] & U>
>
