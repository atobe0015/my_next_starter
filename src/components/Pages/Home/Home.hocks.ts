export type UseHockReturn = ReturnType<typeof useHock>

export const useHock = () => {
  const userName = 'あなたの名前'
  return {
    userName
  }
}
