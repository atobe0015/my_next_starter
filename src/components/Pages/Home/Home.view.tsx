import React from 'react'
import type { UseHockReturn } from './Home.hocks'

export const View = (props: UseHockReturn) => {
  return <div className="text-center">welcome, {props.userName}</div>
}
