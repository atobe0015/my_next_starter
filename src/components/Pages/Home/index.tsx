import { View } from "./Home.view";
import { useHock } from "./Home.hocks";

export const Home = () => {
  const props = useHock();
  return <View {...props} />;
};

export default Home;
