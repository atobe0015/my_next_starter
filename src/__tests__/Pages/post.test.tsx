import React from 'react'
import Home from '@/pages/index'
import { render } from '../Utils'

test('サンプルスナップショットテスト', () => {
  const { container } = render(<Home />)
  expect(container).toMatchSnapshot()
})
