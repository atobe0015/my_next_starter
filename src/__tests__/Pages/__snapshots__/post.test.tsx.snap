// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`投稿テスト 1`] = `
<div>
  <div
    class="grid place-content-center h-screen"
  >
    <div
      class="text-center"
    >
      welcome, 
      あなたの名前
    </div>
  </div>
</div>
`;
